parsec/core/fs/storage/manifest_storage.py

```diff
@@ -269,37 +269,46 @@ async def _ensure_manifest_persistent(self, entry_id: EntryID) -> None:
             if entry_id not in self._cache_ahead_of_localdb:
                 return
 
-            # Safely get the manifest
+            # Safely get the manifest and other information
             manifest = self._cache[entry_id]
-
-            # Dump and decrypt the manifest
-            ciphered = manifest.dump_and_encrypt(self.device.local_symkey)
-
-            # Insert into the local database
-            cursor.execute(
-                """INSERT OR REPLACE INTO vlobs (vlob_id, blob, need_sync, base_version, remote_version)
-                VALUES (
+            pending_chunks_ids = [
```

Comment:
The behavior has changed here:
- in the old code `self._cache_ahead_of_localdb[entry_id]` was computed at the end of the operations
- in the new code, `self._cache_ahead_of_localdb[entry_id]` is computed at the beginning of the operations

On top of that, the old code blocked the trio loop (so no concurrent operations could change `self._cache_ahead_of_localdb[entry_id]` in our back) and now we are doing our stuff in another thread (so `self._cache_ahead_of_localdb[entry_id]` can be modified at any time)

I guess we need more comment explaining the intent here ;-)

Reply:
That's exactly right :grin: 

The overall idea is explained here: https://github.com/Scille/parsec-cloud/issues/1715

Also note the comment below, line 310:
> # Tag entry as up-to-date only if no new manifest has been written in the meantime